### PR TITLE
Add the grep command

### PR DESCRIPTION
**Pull request by nounoursheureux**
See redox-os/coreutils#32

### CODE REVIEW

File: src/bin/grep.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    let mut files = Vec::with_capacity(args.len()-1);
+    for arg in args {
+        if arg.starts_with("-") {
+            match arg.as_str() {
+                "-h" | "--help" => {
+                    stdout.writeln(MAN_PAGE.as_bytes()).try(&mut stderr);
+                },
+                "-n" | "--line-number" => flags.line_numbers = true,
+                _ => {
+                    fail(&format!("Unkown option: {}", &arg), &mut stderr);
```
Comment (Ticki) [inline src/bin/grep.rs]: No need for formatting, just sequentially print.

File: src/bin/grep.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                }
+            }
+        } else if &pattern == "" {
+            pattern = arg.clone();
+        } else {
+            match File::open(&Path::new(&arg)) {
+                Ok(f) => files.push(f),
+                Err(e) => {
+                    fail(&format!("Error opening {}: {}", &arg, e.description()), &mut stderr);
```
Comment (Ticki) [inline src/bin/grep.rs]: Same here.

File: src/bin/grep.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                }
+            }
+        }
+    }
+
+    if &pattern == "" {
```
Comment (Ticki) [inline src/bin/grep.rs]: `.is_empty()`

**Comment by Ticki** [comment]
Thank you!

**Comment by nounoursheureux** [comment]
This should be fixed.

**Comment by Ticki** [comment]
Thanks!

**Pull request merged**